In [1]:
#Andrew Weis
#Data Skills Assignment 7
#10 December 2019

import pandas as pd
import numpy as np
import bokeh
import matplotlib.pyplot as plt
from bokeh.plotting import figure, show
from bokeh.layouts import gridplot
from ipywidgets import interact, interact_manual


In [2]:
from bokeh.io import output_notebook
output_notebook()

Loading BokehJS ...

# Start with Assignment 6 code

In [3]:
PATH='/Users/andrewweis/Documents/GitHub/DataSkills/'
FOLDER='assignment-6-awastro55/'

In [4]:
#Copied this function in part from assignment 1
def read_df(files):
    dfs=[]
    for f in files:
        df=pd.read_csv(PATH+FOLDER+f)
        dfs.append(df)
        
    return dfs

In [5]:
#Copied this function from assignment 5.  
def csv_gen(sets,base):
    return base.format(sets)



In [6]:
#Find data types
#Copied this function from assignment 5.  
def check(df):
    return df.dtypes



In [7]:
#Modeled this on similar code from assignment 5.  
csvbase='EducationDataPortal_12.02.2019_{}.csv'
datasets=['datadictionary','institutions','level_of_study','type_of_freshman']
csvlist=[csv_gen(item, csvbase) for item in datasets]
csvlist

['EducationDataPortal_12.02.2019_datadictionary.csv',
 'EducationDataPortal_12.02.2019_institutions.csv',
 'EducationDataPortal_12.02.2019_level_of_study.csv',
 'EducationDataPortal_12.02.2019_type_of_freshman.csv']

In [8]:
education=read_df(csvlist)
education #Gets 4 data frames.  

[                 variable                                       label  \
 0                  unitid                              Unit ID number   
 1          number_applied                        Number of applicants   
 2         number_admitted                        Number of admissions   
 3   sat_crit_read_25_pctl  SAT critical reading 25th percentile score   
 4   sat_crit_read_75_pctl  SAT critical reading 75th percentile score   
 5        sat_math_25_pctl              SAT math 25th percentile score   
 6        sat_math_75_pctl              SAT math 75th percentile score   
 7   act_composite_25_pctl         ACT composite 25th percentile score   
 8   act_composite_75_pctl         ACT composite 75th percentile score   
 9     act_english_25_pctl           ACT english 25th percentile score   
 10    act_english_75_pctl           ACT english 75th percentile score   
 11       act_math_25_pctl              ACT math 25th percentile score   
 12       act_math_75_pctl            

In [9]:
check(education[0])

variable        object
label           object
description     object
source          object
source2        float64
dtype: object

In [10]:
check(education[1])

year                       int64
unitid                     int64
inst_name                 object
state_name                object
number_applied           float64
number_admitted          float64
sat_crit_read_25_pctl    float64
sat_crit_read_75_pctl    float64
sat_math_25_pctl         float64
sat_math_75_pctl         float64
act_composite_25_pctl    float64
act_composite_75_pctl    float64
act_english_25_pctl      float64
act_english_75_pctl      float64
act_math_25_pctl         float64
act_math_75_pctl         float64
inst_control              object
institution_level         object
dtype: object

In [11]:
check(education[2])

year                int64
unitid              int64
inst_name          object
state_name         object
level_of_study     object
enrollment_fall     int64
dtype: object

In [12]:
check(education[3])

year                    int64
unitid                  int64
inst_name              object
state_name             object
state_of_residence     object
type_of_freshman       object
enrollment_fall       float64
dtype: object

In [13]:
#Separate data frames
admit=education[1]
level=education[2]
freshman=education[3]

In [53]:
admit['acceptance_rate']=admit['number_admitted']/admit['number_applied']
admit

year  unitid                                          inst_name  \
0    2016  160959                            College of the Atlantic   
1    2016  160977                                      Bates College   
2    2016  161004                                    Bowdoin College   
3    2016  161086                                      Colby College   
4    2016  161217                     University of Maine at Augusta   
5    2016  161226                  University of Maine at Farmington   
6    2016  161235                   University of Maine at Fort Kent   
7    2016  161244                     University of Maine at Machias   
8    2016  161253                                University of Maine   
9    2016  161280          University of Maine-System Central Office   
10   2016  161299                             Maine Maritime Academy   
11   2016  161341                University of Maine at Presque Isle   
12   2016  161457                          University of New England   
13   2016  161509                               Maine College of Art   
14   2016  161518                    Saint Joseph's College of Maine   
15   2016  161554                       University of Southern Maine   
16   2016  161563                                     Thomas College   
17   2016  161572                                      Unity College   
18   2016  462044  Institute for Doctoral Studies in the Visual Arts   
19   2016  487524                                  Husson University   
20   2015  160959                            College of the Atlantic   
21   2015  160977                                      Bates College   
22   2015  161004                                    Bowdoin College   
23   2015  161086                                      Colby College   
24   2015  161165                                  Husson University   
25   2015  161217                     University of Maine at Augusta   
26   2015  161226                  University of Maine at Farmington   
27   2015  161235                   University of Maine at Fort Kent   
28   2015  161244                     University of Maine at Machias   
29   2015  161253                                University of Maine   
..    ...     ...                                                ...   
318  2002  161518                              SAINT JOSEPHS COLLEGE   
319  2002  161554                       UNIVERSITY OF SOUTHERN MAINE   
320  2002  161563                                     THOMAS COLLEGE   
321  2002  161572                                      UNITY COLLEGE   
322  2002  373827               NEW ENGLAND SCHOOL OF COMMUNICATIONS   
323  2002  407230                        BANGOR THEOLOGICAL SEMINARY   
324  2001  160959                            COLLEGE OF THE ATLANTIC   
325  2001  160968                        BANGOR THEOLOGICAL SEMINARY   
326  2001  160977                                      BATES COLLEGE   
327  2001  161004                                    BOWDOIN COLLEGE   
328  2001  161086                                      COLBY COLLEGE   
329  2001  161147                           FAITH SCHOOL OF THEOLOGY   
330  2001  161165                                     HUSSON COLLEGE   
331  2001  161217                     UNIVERSITY OF MAINE AT AUGUSTA   
332  2001  161226                  UNIVERSITY OF MAINE AT FARMINGTON   
333  2001  161235                   UNIVERSITY OF MAINE AT FORT KENT   
334  2001  161244                     UNIVERSITY OF MAINE AT MACHIAS   
335  2001  161253                                UNIVERSITY OF MAINE   
336  2001  161280          UNIVERSITY OF MAINE SYSTEM CENTRAL OFFICE   
337  2001  161299                             MAINE MARITIME ACADEMY   
338  2001  161341                UNIVERSITY OF MAINE AT PRESQUE ISLE   
339  2001  161448                  NEW ENGLAND BAPTIST BIBLE COLLEGE   
340  2001  161457        UNIVERSITY OF NEW ENGLAND-UNIVERSITY CAMPUS   
341  2001  161509                            

In [87]:
#Summary stats
print(np.mean(admit['acceptance_rate']))
print(np.std(admit['acceptance_rate']))
print(np.min(admit[admit['acceptance_rate']>0]['acceptance_rate']))
print(np.max(admit['acceptance_rate']))
admit.describe()

0.6787589313918648
0.22986986106376603
0.14840417708486542
0.9805352798053528


year         unitid  number_applied  number_admitted  \
count   348.000000     348.000000      262.000000       262.000000   
mean   2008.379310  178913.146552     2291.335878      1309.492366   
std       4.643464   64554.791849     2394.662123      1579.722898   
min    2001.000000  160959.000000        0.000000         0.000000   
25%    2004.000000  161165.000000      472.000000       364.250000   
50%    2008.000000  161280.000000     1259.500000      1006.500000   
75%    2012.000000  161518.000000     4110.500000      1397.000000   
max    2016.000000  487524.000000    12952.000000     11625.000000   

       sat_crit_read_25_pctl  sat_crit_read_75_pctl  sat_math_25_pctl  \
count             163.000000             163.000000        163.000000   
mean              480.398773             583.558282        482.607362   
std                81.236550              75.108407         83.836145   
min               370.000000             480.000000        350.000000   
25%               430.000000             530.000000        430.000000   
50%               450.000000             560.000000        460.000000   
75%               480.000000             590.000000        490.000000   
max               680.000000             760.000000        680.000000   

       sat_math_75_pctl  act_composite_25_pctl  act_composite_75_pctl  \
count        163.000000             116.000000             116.000000   
mean         585.276074              20.181034              25.362069   
std           74.150022               4.390898               3.786467   
min          470.000000              15.000000              19.000000   
25%          535.000000              17.000000              23.000000   
50%          570.000000              19.000000              25.000000   
75%          600.000000              21.000000              27.000000   
max          760.000000              31.000000              33.000000   

       act_english_25_pctl  act_english_75_pctl  act_math_25_pctl  \
count            91.000000            91.000000         91.000000   
mean             19.747253            25.758242         19.923077   
std               5.338734             4.648394          4.351834   
min              12.000000            18.000000         15.000000   
25%              16.000000            22.500000         17.000000   
50%              18.000000            24.000000         18.000000   
75%              20.500000            28.500000         21.000000   
max              31.000000            35.000000         29.000000   

       act_math_75_pctl  acceptance_rate  
count         91.000000       257.000000  
mean          25.989011         0.678759  
std            3.920018         0.230318  
min           19.000000         0.000000  
25%           23.000000         0.640327  
50%           25.000000         0.771151  
75%           27.000000         0.826360  
max           34.000000         0.980535

In [18]:
colleges=admit.reset_index()['inst_name'].unique()
for college in colleges:
    college=college.lower()
    
colleges

array(['College of the Atlantic', 'Bates College', 'Bowdoin College',
       'Colby College', 'University of Maine at Augusta',
       'University of Maine at Farmington',
       'University of Maine at Fort Kent',
       'University of Maine at Machias', 'University of Maine',
       'University of Maine-System Central Office',
       'Maine Maritime Academy', 'University of Maine at Presque Isle',
       'University of New England', 'Maine College of Art',
       "Saint Joseph's College of Maine", 'University of Southern Maine',
       'Thomas College', 'Unity College',
       'Institute for Doctoral Studies in the Visual Arts',
       'Husson University', 'New England School of Communications',
       'Bangor Theological Seminary',
       'University of Maine System Central Office', 'Husson College',
       'New England Baptist Bible College', 'Saint Josephs College',
       'COLLEGE OF THE ATLANTIC', 'BANGOR THEOLOGICAL SEMINARY',
       'BATES COLLEGE', 'BOWDOIN COLLEGE', 'COLBY C

In [17]:
def plot_one_rate(college, color='red', size=8):
    plot = figure(title='Acceptance Rates Over Time For '+college, x_axis_label='Year', 
              y_axis_label='Rate')
    plot.circle(admit[admit['inst_name']==college]['year'], 
                 admit[admit['inst_name']==college]['acceptance_rate'],
              legend=college, fill_color=color, size=size)
    #plot2 = figure(title='Acceptance Rates Over Time For '+college2, x_axis_label='Year', 
     #         y_axis_label='Rate', x
        
    #show(plot)
    return(plot)

In [18]:
@interact(college=colleges)
def make_one_plot(college=colleges[2]):
    plot=plot_one_rate(college)
    show(plot)

interactive(children=(Dropdown(description='college', index=2, options=('College of the Atlantic', 'Bates Coll…

In [19]:
#Compare rates for two colleges.
def plot_rate(college1, college2, color='red', size=8):
    plot1 = figure(title='Acceptance Rates Over Time For '+college1, x_axis_label='Year', 
              y_axis_label='Rate')
    plot1.circle(admit[admit['inst_name']==college1]['year'], 
                 admit[admit['inst_name']==college1]['acceptance_rate'],
              legend=college1, fill_color=color, size=size)
    plot2 = figure(title='Acceptance Rates Over Time For '+college2, x_axis_label='Year', 
              y_axis_label='Rate', x_range=plot1.x_range, y_range=plot1.y_range)
    plot2.circle(admit[admit['inst_name']==college2]['year'], 
                 admit[admit['inst_name']==college2]['acceptance_rate'],
              legend=college2, fill_color=color, size=size)
    plot=gridplot([[plot1],[plot2]])
    #show(plot)
    return(plot)

In [20]:
@interact(college1=colleges, college2=colleges)#colleges=admit['inst_name'])
def make_rate_plot(college1=colleges[2],college2=colleges[3]):
    plot=plot_rate(college1, college2)
    show(plot)

interactive(children=(Dropdown(description='college1', index=2, options=('College of the Atlantic', 'Bates Col…

In [22]:
#Plot by residence 
def plot_residence(college,df):
    counts=df[df['inst_name']==college]['state_of_residence'].value_counts()
    plot=figure(title='Residents by State For '+college, x_axis_label='State', 
              y_axis_label='Number', 
                x_range=df.reset_index()['state_of_residence'].unique())
    plot.vbar(x=df['state_of_residence'], top=counts, width=0.4)
    #show(plot)
    return plot

In [27]:
@interact(college=colleges)
def make_res_plot(college=colleges[2]):
    plot=plot_residence(college,freshman)
    show(plot)

interactive(children=(Dropdown(description='college', index=2, options=('College of the Atlantic', 'Bates Coll…

In [28]:
from sklearn.linear_model import LinearRegression
model = LinearRegression(fit_intercept=False)

In [69]:
#Make a categorical variable based on acceptance rate.  
admit["Selectivity"]=pd.cut(admit['acceptance_rate'], bins=[0,.20,.40,.60,1.0],
                           include_lowest=True,
                           labels=['Highly Selective', 'Selective',
                                  'Somewhat Selective', 'Not Selective'])

In [70]:
admit

year  unitid                                          inst_name  \
0    2016  160959                            College of the Atlantic   
1    2016  160977                                      Bates College   
2    2016  161004                                    Bowdoin College   
3    2016  161086                                      Colby College   
4    2016  161217                     University of Maine at Augusta   
5    2016  161226                  University of Maine at Farmington   
6    2016  161235                   University of Maine at Fort Kent   
7    2016  161244                     University of Maine at Machias   
8    2016  161253                                University of Maine   
9    2016  161280          University of Maine-System Central Office   
10   2016  161299                             Maine Maritime Academy   
11   2016  161341                University of Maine at Presque Isle   
12   2016  161457                          University of New England   
13   2016  161509                               Maine College of Art   
14   2016  161518                    Saint Joseph's College of Maine   
15   2016  161554                       University of Southern Maine   
16   2016  161563                                     Thomas College   
17   2016  161572                                      Unity College   
18   2016  462044  Institute for Doctoral Studies in the Visual Arts   
19   2016  487524                                  Husson University   
20   2015  160959                            College of the Atlantic   
21   2015  160977                                      Bates College   
22   2015  161004                                    Bowdoin College   
23   2015  161086                                      Colby College   
24   2015  161165                                  Husson University   
25   2015  161217                     University of Maine at Augusta   
26   2015  161226                  University of Maine at Farmington   
27   2015  161235                   University of Maine at Fort Kent   
28   2015  161244                     University of Maine at Machias   
29   2015  161253                                University of Maine   
..    ...     ...                                                ...   
318  2002  161518                              SAINT JOSEPHS COLLEGE   
319  2002  161554                       UNIVERSITY OF SOUTHERN MAINE   
320  2002  161563                                     THOMAS COLLEGE   
321  2002  161572                                      UNITY COLLEGE   
322  2002  373827               NEW ENGLAND SCHOOL OF COMMUNICATIONS   
323  2002  407230                        BANGOR THEOLOGICAL SEMINARY   
324  2001  160959                            COLLEGE OF THE ATLANTIC   
325  2001  160968                        BANGOR THEOLOGICAL SEMINARY   
326  2001  160977                                      BATES COLLEGE   
327  2001  161004                                    BOWDOIN COLLEGE   
328  2001  161086                                      COLBY COLLEGE   
329  2001  161147                           FAITH SCHOOL OF THEOLOGY   
330  2001  161165                                     HUSSON COLLEGE   
331  2001  161217                     UNIVERSITY OF MAINE AT AUGUSTA   
332  2001  161226                  UNIVERSITY OF MAINE AT FARMINGTON   
333  2001  161235                   UNIVERSITY OF MAINE AT FORT KENT   
334  2001  161244                     UNIVERSITY OF MAINE AT MACHIAS   
335  2001  161253                                UNIVERSITY OF MAINE   
336  2001  161280          UNIVERSITY OF MAINE SYSTEM CENTRAL OFFICE   
337  2001  161299                             MAINE MARITIME ACADEMY   
338  2001  161341                UNIVERSITY OF MAINE AT PRESQUE ISLE   
339  2001  161448                  NEW ENGLAND BAPTIST BIBLE COLLEGE   
340  2001  161457        UNIVERSITY OF NEW ENGLAND-UNIVERSITY CAMPUS   
341  2001  161509                            

# Now, the models

In [71]:
Z=admit[['sat_math_75_pctl','act_composite_75_pctl', 'Selectivity']].dropna()
X=Z[['sat_math_75_pctl','act_composite_75_pctl']]
Y=Z['Selectivity']
X

sat_math_75_pctl  act_composite_75_pctl
3               745.0                   33.0
6               475.0                   22.0
8               600.0                   26.0
10              580.0                   25.0
12              580.0                   26.0
14              500.0                   23.0
15              540.0                   25.0
19              540.0                   23.0
23              740.0                   32.0
27              500.0                   19.0
29              610.0                   28.0
31              590.0                   26.0
33              590.0                   29.0
35              540.0                   26.0
36              550.0                   25.0
41              540.0                   23.0
45              720.0                   32.0
46              530.0                   21.0
49              500.0                   19.0
51              600.0                   27.0
53              580.0                   24.0
55              580.0                   26.0
58              550.0                   24.0
66              750.0                   33.0
67              720.0                   32.0
68              550.0                   21.0
71              500.0                   27.0
72              520.0                   25.0
73              610.0                   26.0
75              580.0                   24.0
..                ...                    ...
209             540.0                   23.0
210             510.0                   22.0
217             720.0                   31.0
222             520.0                   25.0
223             600.0                   25.0
225             580.0                   23.0
229             580.0                   20.0
231             540.0                   23.0
232             530.0                   25.0
239             710.0                   31.0
244             540.0                   25.0
245             600.0                   26.0
247             580.0                   22.0
251             570.0                   20.0
253             550.0                   23.0
254             520.0                   24.0
261             710.0                   31.0
266             540.0                   21.0
267             600.0                   27.0
269             590.0                   26.0
273             560.0                   20.0
274             550.0                   20.0
275             550.0                   24.0
288             520.0                   23.0
295             560.0                   20.0
310             520.0                   26.0
313             610.0                   24.0
317             540.0                   26.0
328             710.0                   30.0
341             530.0                   24.0

[116 rows x 2 columns]

In [72]:
Y

3      Highly Selective
6         Not Selective
8         Not Selective
10        Not Selective
12        Not Selective
14        Not Selective
15        Not Selective
19        Not Selective
23            Selective
27        Not Selective
29        Not Selective
31        Not Selective
33        Not Selective
35        Not Selective
36        Not Selective
41        Not Selective
45            Selective
46        Not Selective
49        Not Selective
51        Not Selective
53        Not Selective
55        Not Selective
58        Not Selective
66     Highly Selective
67            Selective
68        Not Selective
71        Not Selective
72        Not Selective
73        Not Selective
75        Not Selective
             ...       
209       Not Selective
210       Not Selective
217           Selective
222       Not Selective
223       Not Selective
225       Not Selective
229       Not Selective
231       Not Selective
232       Not Selective
239           Selective
244       Not Se

In [73]:
X.shape

(116, 2)

In [74]:
Y.shape

(116,)

# Try the SVC model, as in class. 

In [99]:
from sklearn.model_selection import train_test_split

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.20, random_state=123) 

In [89]:
print(X_train.shape)
print(Y_train.shape)

print(X_test.shape)
print(Y_test.shape)

(81, 2)
(81,)
(35, 2)
(35,)


In [100]:
from sklearn.svm import SVC
model = SVC(gamma='auto')
model.fit(X_train, Y_train)
predict = model.predict(X_test)

In [91]:
predict

array(['Not Selective', 'Not Selective', 'Highly Selective',
       'Not Selective', 'Not Selective', 'Not Selective', 'Not Selective',
       'Not Selective', 'Not Selective', 'Not Selective', 'Not Selective',
       'Not Selective', 'Not Selective', 'Not Selective', 'Not Selective',
       'Selective', 'Not Selective', 'Not Selective', 'Selective',
       'Not Selective', 'Selective', 'Not Selective', 'Not Selective',
       'Selective', 'Not Selective', 'Not Selective', 'Not Selective',
       'Highly Selective', 'Not Selective', 'Not Selective', 'Selective',
       'Not Selective', 'Not Selective', 'Not Selective', 'Not Selective'],
      dtype=object)

In [60]:
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error

In [61]:
r2_score(Y_test, predict)
mean_squared_error(Y_test, predict)

0.0324088046360858

In [92]:
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

In [101]:
accuracy_score(Y_test, predict)

0.875

In [94]:
confusion_matrix(Y_test, predict)

array([[ 2,  2,  0],
       [ 0, 25,  0],
       [ 0,  1,  5]])

In [95]:
print(classification_report(Y_test, predict))

                  precision    recall  f1-score   support

Highly Selective       1.00      0.50      0.67         4
   Not Selective       0.89      1.00      0.94        25
       Selective       1.00      0.83      0.91         6

        accuracy                           0.91        35
       macro avg       0.96      0.78      0.84        35
    weighted avg       0.92      0.91      0.91        35



# Test Harness--Model Selection

In [96]:
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import StratifiedKFold

In [97]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
from sklearn.naive_bayes import GaussianNB

models = [('Dec Tree', DecisionTreeClassifier()), 
          ('Lin Disc', LinearDiscriminantAnalysis()), ('Gauss', GaussianNB()), 
          ('SVC', SVC(gamma='auto'))]

In [102]:
results = []

for name, model in models:
    kf = StratifiedKFold(n_splits=10, random_state=123)
    res = cross_val_score(model, X_train, Y_train, cv=kf, scoring='accuracy') #Not kfold
    res_mean = round(res.mean(), 4) #Constraints, tougher with bigger data. 
    res_std  = round(res.std(), 4)
    results.append((name, res_mean, res_std))
    
results 

//anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py:657: Warning: The least populated class in y has only 5 members, which is too few. The minimum number of members in any class cannot be less than n_splits=10.
  % (min_groups, self.n_splits)), Warning)
//anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py:657: Warning: The least populated class in y has only 5 members, which is too few. The minimum number of members in any class cannot be less than n_splits=10.
  % (min_groups, self.n_splits)), Warning)
//anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py:657: Warning: The least populated class in y has only 5 members, which is too few. The minimum number of members in any class cannot be less than n_splits=10.
  % (min_groups, self.n_splits)), Warning)
//anaconda3/lib/python3.7/site-packages/sklearn/model_selection/_split.py:657: Warning: The least populated class in y has only 5 members, which is too few. The minimum num

[('Dec Tree', 0.97, 0.064),
 ('Lin Disc', 0.97, 0.064),
 ('Gauss', 0.99, 0.03),
 ('SVC', 0.97, 0.064)]

# Try the Gaussian model, see if it predicts better than SVC. 

In [103]:
Gaussmodel=GaussianNB()
Gaussmodel.fit(X_train,Y_train)
Gausspredict=Gaussmodel.predict(X_test)

In [104]:
Gausspredict

array(['Not Selective', 'Not Selective', 'Highly Selective',
       'Not Selective', 'Not Selective', 'Not Selective', 'Not Selective',
       'Not Selective', 'Not Selective', 'Highly Selective',
       'Not Selective', 'Selective', 'Not Selective', 'Highly Selective',
       'Not Selective', 'Selective', 'Not Selective', 'Not Selective',
       'Selective', 'Not Selective', 'Selective', 'Not Selective',
       'Not Selective', 'Selective'], dtype='<U16')

In [105]:
accuracy_score(Y_test, Gausspredict)

1.0

In [106]:
confusion_matrix(Y_test, Gausspredict)

array([[ 3,  0,  0],
       [ 0, 16,  0],
       [ 0,  0,  5]])

In [107]:
print(classification_report(Y_test, Gausspredict))

                  precision    recall  f1-score   support

Highly Selective       1.00      1.00      1.00         3
   Not Selective       1.00      1.00      1.00        16
       Selective       1.00      1.00      1.00         5

        accuracy                           1.00        24
       macro avg       1.00      1.00      1.00        24
    weighted avg       1.00      1.00      1.00        24



# Final Thoughts

I used the education data from assignment 6; most of the first half of this code is taken directly from assignment 6.  I used 75% SAT math and ACT composite scores to predict selectivity of colleges, defining 20% admission or smaller as highly selective, 20-40% as selective, 40-60% as somewhat selective, and higher than 60% as not selective.  I expected from the summary statistics and the models that SAT and ACT scores are reasonably good predictors of college selectiveness.  I used accuracy score to find out how many colleges the model predicted correctly.  The SVC method predicted 87.5% correctly, but the Gaussian method, which the test harness had said was the most accurate, predicted everything correctly.  I would not expect that to happen in most cases, but the Gaussian does appear to be the best model in this circumstance, in terms of accuracy.  